### ISSUE TITLE
🐛 Bug Report: Stops recognising new songs after 2

### ISSUE THREAD

**Issue by Kayrah87** [whatsnowplaying/whats-now-playing#994]
### 📜 Description

It appears to stop picking up file system changes after the first two tracks. It becomes stuck on the second track and will not proceed

### 👟 Reproduction steps

When I load a DJuced playlist I play the first track in deck 1. The template shows in in OBS and the twitchbot receives the first track info in the chat. 
I load the second track into deck 2. When I hit play the template changes in OBS, and the new track title is sent to twitch chat. 
I load the third track into deck 1, but when I hit play nothing changes on the template and no track info is sent to twitch chat.
I have checked playing.txt and can see that DJuced is updating the file accurately. I have also paused Onedrive as I suspected that might be causing issues, but the issue persists

### 👍 Expected behavior

The template updates and the twicth chat receives the new track info

### 👎 Actual Behavior

It remains on the second track info, and does not update either the template or the twitch chat.

### 💻 Operating system

Windows

### 💻 Chipset/CPU

Intel

### 💻 Input source

Other

### 🧱 Your Environment

I use DJuced for local tracks. I use OBS straight to Twitch. I have a number of visual sources in the scene but I doubt they have anything to do with it. 
I run Discord in the background and use VRChat in the foreground for my vtube avatar. 
Note that the issue persists when VRChat is closed. 
I have paused Onedrive syncing but the issue persists.
[debug.log](https://github.com/whatsnowplaying/whats-now-playing/files/13956405/debug.log)

**Comment by aw-was-here** [comment]
I agree, it definitely sounds like a bug in file detection change.  I'll see if I can reproduce.  It has been a while since I fired up DJuced so maybe something has changed with it.

**Comment by Kayrah87** [comment]
For additional info, I have set up Traktor as my main DJ app to test functionality of the now playing software, and that works absolutely fine. The difference obviously being icecast vs file watcher, which probably helps narrow it down. Hope that helps.

**Comment by aw-was-here** [comment]
Hmm. Trying this on Win 11 w/DJuced 6.0.1 and it is definitely picking up the 3rd track. I'm working on getting a release out and have a few fixes for other things so it may be an existing bug that is already fixed.

**Comment by aw-was-here** [comment]
I'm hoping this one is fixed in 4.2.0 RC1, esp since I can't reproduce this one.

**Comment by Kayrah87** [comment]
Unfortunately not. It works absolutely fine when the system is under no load, but then when I am streaming and DJing it will pick up maybe 1 song in 5.
I have also switched to DJing on a Mac and streaming from another PC so I’ve got the mac running the server for NP and the PC connecting to it over the network. It has exactly the same problems.

Part of me feels now like it’s something to do with a combination of the PC load, the speed of the connection, and maybe a timeout which means that it can’t get the track info in time so just gives up maybe? Just a fleeting guess.

From: RoughTradeRadio ***@***.***>
Date: Thursday, 17 October 2024 at 17:44
To: whatsnowplaying/whats-now-playing ***@***.***>
Cc: Kayleigh Whitehurst ***@***.***>, Mention ***@***.***>
Subject: Re: [whatsnowplaying/whats-now-playing] 🐛 Bug Report: Stops recognising new songs after 2 (Issue #994)

@Kayrah87<https://github.com/Kayrah87> Did you manage to get this resolved or find a workaround?

**Comment by aw-was-here** [comment]
Try 4.2.0, which was just released.

**Comment by aw-was-here** [comment]
It _may_ be useful to upgrade anyway because the new version has config import/export....

**Comment by aw-was-here** [comment]
Hmm. I'll see if I missed another spot where stuff is running that shouldn't be.  Obviously if artist extras is enabled that's going to take up some time but if you have that off and write delay in the general pref is set to 1, something isn't getting short circuited.

**Comment by aw-was-here** [comment]
Just a dump of my double check today.  

Contents of my History directory:

<img width="377" height="148" alt="Image" src="https://github.com/user-attachments/assets/f2e84826-815b-478f-a8d9-061828803b09" />

Portion of my debug log. I've got notifications, real time set lists, and charts from 5.x turned on.  I didn't have kick or twitch turned on because they just fill the log with other stuff making it hard to see.

<img width="1170" height="474" alt="Image" src="https://github.com/user-attachments/assets/28360733-50aa-4952-ac4b-c563ce5d3ed3" />

Completed the whole cycle in 3 seconds. The one that matters the most is this line:
 ```2025-08-30T14:40:55-0700 DEBUG 7853 trackpoll/TrackPoll db:write_to_metadb:210 Adding record with Elastica/Connection
```

That line is when everything starts firing off as every subsystem depends upon that database record. So all of that happened at 2 seconds, including the 1 second delay timer from the main settings page.

Now when I have Twitch chat turned on, it waits until the chat delay timer setting (Streaming & Chat -> Twitch -> Chat -> Announce Delay) seconds and then fires off a message like clock work. (Setting that too low, BTW, will result in messages getting sent to chat before the audio/video actually gets to the users since Twitch has to re-encode your stream, so be aware of that).

Ran this experiment over and over and get pretty much the same results every time.  One thing that does seem to be happening is that I'm not convinced that VDJ is honoring the write time on the History file.  I was using 5 seconds but it seemed to be longer.  I know they've _greatly_ increased their file IO over the past year or so and that might be a cause of that.

The one thing that _might_ be causing issues here is the system trying to resolve your hostname/ip address. I'll see if I can make that smarter, but it should be pretty quick.
